src/interfaces/pyo3/hybrid_gpsw_aes.rs

```diff
@@ -0,0 +1,254 @@
+            &symmetric_key,
+            &uid,
+            block_number_value as usize,
+            &plaintext_bytes,
+        )
+        .map_err(|e| PyTypeError::new_err(format!("Error encrypting block: {e}")))?;
+
+    Ok(ciphertext)
+}
+
+/// Symmetrically Decrypt encrypted data in a block.
+#[pyfunction]
+pub fn decrypt_hybrid_block(
+    symmetric_key_bytes: Vec<u8>,
+    uid_bytes: Option<Vec<u8>>,
+    block_number: Option<usize>,
+    encrypted_bytes: Vec<u8>,
+) -> PyResult<Vec<u8>> {
+    //
+    // Check `user_decryption_key_bytes` input param and store it locally
+    if symmetric_key_bytes.len() != 32 {
+        return Err(PyTypeError::new_err("Symmetric key must be 32-bytes long"));
+    }
+
+    //
+    // Check `encrypted_bytes` input param and store it locally
+    if encrypted_bytes.is_empty() {
+        return Err(PyTypeError::new_err("Encrypted value is empty"));
+    }
+
+    //
+    // Parse symmetric key
+    let symmetric_key =
+        <Aes256GcmCrypto as SymmetricCrypto>::Key::try_from_bytes(symmetric_key_bytes)
+            .map_err(|e| PyTypeError::new_err(format!("Deserialize symmetric key failed: {e}")))?;
+
+    //
+    // Parse other input params
+    let uid = uid_bytes.map_or(vec![], |v| v.to_vec());
+    let block_number_value = block_number.unwrap_or(0);
+
+    //
+    // Decrypt block
+    let cleartext =
+        core_decrypt_hybrid_block::<Gpsw<Bls12_381>, Aes256GcmCrypto, MAX_CLEAR_TEXT_SIZE>(
+            &symmetric_key,
+            &uid,
+            block_number_value as usize,
+            &encrypted_bytes,
+        )
+        .map_err(|e| PyTypeError::new_err(format!("Error encrypting block: {e}")))?;
+
+    Ok(cleartext)
+}
+
+#[pyfunction]
+pub fn encrypt(
+    metadata_bytes: Vec<u8>,
+    policy_bytes: Vec<u8>,
+    attributes_bytes: Vec<u8>,
+    public_key_bytes: Vec<u8>,
+    plaintext: Vec<u8>,
+) -> PyResult<Vec<u8>> {
+    let metadata: Metadata = serde_json::from_slice(&metadata_bytes)
+        .map_err(|e| PyTypeError::new_err(format!("Error deserializing metadata: {e}")))?;
+
+    let header = encrypt_hybrid_header(
+        metadata_bytes,
+        policy_bytes,
+        attributes_bytes,
+        public_key_bytes,
+    )?;
+
+    let ciphertext = encrypt_hybrid_block(header.0, Some(metadata.uid), None, plaintext)?;
+
+    // Encrypted value is composed of: HEADER_LEN (4 bytes) | HEADER | AES_DATA
+    let mut encrypted = Vec::<u8>::with_capacity(4 + header.1.len() + ciphertext.len());
+    encrypted.extend_from_slice(&u32::to_be_bytes(header.1.len() as u32));
+    encrypted.extend_from_slice(&header.1);
+    encrypted.extend_from_slice(&ciphertext);
+    Ok(encrypted)
+}
+
+#[pyfunction]
+pub fn decrypt(user_decryption_key_bytes: Vec<u8>, encrypted_bytes: Vec<u8>) -> PyResult<Vec<u8>> {
+    let header_size = get_encrypted_header_size(encrypted_bytes.clone())?;
+    let header = encrypted_bytes[4..4 + header_size as usize].to_vec();
```

Comment:
It would be easier to not expose in Python the `encrypt_hybrid_*` and `decrypt_hybrid_*` but I believe both functions could be used separately to encypt large amount of data (it this case happens...).

But you're right, I added 2 decrypt-functions taking slice references as inputs instead.